core/src/executor/aggregate.rs

```diff
@@ -138,28 +134,17 @@ async fn group_by_having<'a, T: GStore>(
                             None => Rc::clone(&next),
                         };
                         let filter_context = Some(filter_context);
-                        let aggregated = aggregated.as_ref().map(Rc::clone);
-
-                        check_expr(
-                            storage,
-                            filter_context,
-                            aggregated.as_ref().map(Rc::clone),
-                            having,
-                        )
-                        .await
-                        .map(|pass| pass.then_some((aggregated, next)))
-                        .transpose()
+                        let aggr_rc = aggregated.clone().map(Rc::new);
```

Comment:
im_rc::HashMap internally uses `Rc` so clone operation does not make problem.

we can consider removing `Rc` from `aggregated: Option<Rc<HashMap<&'a Aggregate, Value>>>` in following PRs.